lib/matplotlib/axes/_axes.py

```diff
@@ -834,8 +834,8 @@ def _check_no_units(vals, names):
         # Helper method to check that vals are not unitized
         for val, name in zip(vals, names):
             if not munits._is_natively_supported(val):
-                raise ValueError(f"{name} must be a single scalar value, "
-                                 f"but got {val}")
+                raise ValueError(f"{repr(name)} must be a single scalar value,"
+                                 f" but got {repr(val)}")
```

Comment:
```suggestion
                                 f" but got {val!r}")
```
and so on also works (and is more compact).